python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py

```diff
@@ -636,7 +645,23 @@ def resolve_hasRunMetricsEnabled(self, graphene_info: ResolveInfo):
             return False
 
         run_tags = self.dagster_run.tags
-        return any(get_boolean_tag_value(run_tags.get(tag)) for tag in RUN_METRIC_TAGS)
+
+        # metrics are not kept for runs older than 2 weeks
+        retention_deadline = time.time() - RUN_METRICS_RETENTION_IN_SECONDS  # 2 weeks ago
+        end_time = self._run_record.end_time if self._run_record else None
+        if end_time:
+            # if no end time, run is presumed to be still running
+            if end_time < retention_deadline:
+                return False
+
+        # TODO - deprecate looking for the run_metrics activator tags after 1.9.0
+        eligible_tags = [
+            HAS_RUN_METRICS_TAG,
+            *RUN_METRIC_TAGS,
```

Comment:
removing RUN_METRIC_TAGS now would remove the ability to see metrics from run created until then.